### DISCUSSION TITLE
custodian report not finding anything

### DISCUSSION THREAD

**Discussion by fidelito** [cloud-custodian/cloud-custodian#7337] [General]
When I run custodian report its not finding the instance that meet the policy requirement. I double check in AWS account and it should find a lot of them.

custodian report --profile ACCOUNT NUMBER  -r eu-west-1  -v --format csv  -s /path/reports/ /path/policies/ec2-tag-requirement-check-report.yml
2019-03-15 19:18:41,451: custodian.cache:DEBUG Disabling cache
2019-03-15 19:18:41,451: custodian.commands:DEBUG Loaded file /path/policies/ec2-tag-requirement-check-report.yml. Contains 1 policies
2019-03-15 19:18:41,730: custodian.reports:DEBUG Found 0 records for region eu-west-1
CustodianDate,InstanceId,tag:Name,InstanceType,LaunchTime,VpcId,PrivateIpAddress

ec2-tag-requirement-check-report.yml

vars:
  tag-filters: &tag-compliance-filters
       - "tag:market": present
       - "tag:product": present

policies:
- name: ec2-tag-requirement-check
  resource: ec2
  filters:
     - or: *tag-compliance-filters
  actions:

**Comment by kapilt** [comment]
did you run the policy first? report is against custodian run outputs

**Comment by fidelito** [comment]
Hi Kapil,

I just ran custodian run and then custodian report back to back but custodian report is blank.

Running policy s3-tag-policy.yml against account ACCOUNT NUMBER
15:55:07			2019-03-25 19:55:07,467: custodian.policy:INFO policy: s3-tag-owner-check resource:s3 region:us-east-1 count:26 time:19.66
15:55:08			2019-03-25 19:55:08,225: custodian.actions:INFO sent message:e9f3ffc5-315e-4ec5-9a22-672c77c3d393 policy:s3-tag-owner-check template:default.html count:26
15:55:08			2019-03-25 19:55:08,225: custodian.policy:INFO policy: s3-tag-owner-check action: notify resources: 26 execution_time: 0.75
15:55:09			2019-03-25 19:55:09,485: custodian.cache:DEBUG Disabling cache
15:55:09			2019-03-25 19:55:09,485: custodian.commands:DEBUG Loaded file /dev/fd/63. Contains 1 policies
15:55:09			2019-03-25 19:55:09,778: custodian.reports:DEBUG Found 0 records for region us-east-1
15:55:09			Name,CreationDate


custodian run --profile ${ACCOUNT} --region us-east-1 --cache ${CACHE_DIR}/${POLICY_NAME}/${ACCOUNT}-cache.file --output-dir ${OUTPUT_DIR}/${POLICY_NAME}/${ACCOUNT} <(ACCOUNT=${ACCOUNT} envsubst < ${POLICY_DIR}/${POLICY})

custodian report -v  --profile ${ACCOUNT} --format csv --region us-east-1 --output-dir ${OUTPUT_DIR}/reports/${POLICY_NAME}/${ACCOUNT}/  <(ACCOUNT=${ACCOUNT} envsubst < ${POLICY_DIR}/${POLICY})

any thoughts of what I am doing wrong?

**Comment by jtroberts83** [comment]
The policy run showing matching resources was not for your ec2 policy, it was for a s3 tags policy.
```
s3-tag-owner-check template:default.html count:26
```
Are you seeing items matching your policy **ec2-tag-requirement-check** ?

**Comment by fidelito** [comment]
Hi,

they are the same policy. My apologies for the confusion.

vars:
  tag-filters: &tag-compliance-filters
       - "tag:market": absent
       - "tag:product": absent

policies:
- name: s3-tag-owner-check
  resource: s3
  filters:
    - or: *tag-compliance-filters

**Comment by it-praktyk** [comment]
I observe the same issue.

The custodian_ec2.yml file content

```
policies:

  - name: ec2-tag-compliance
    resource: ec2
    comment: |
      Report on total count of non-compliant instances
    filters:
      - or:
          - "tag:companyname:env": absent
          - "tag:companyname:product": absent
          - "tag:companyname:cc": absent
          - "tag:companyname:owner": absent
```
Output of custodian run
```
custodian run -r eu-west-1 --output-dir . ./custodian_ec2.yml 
2021-07-02 13:12:22,201: custodian.policy:INFO policy:ec2-tag-compliance resource:ec2 region:eu-west-1 count:13 time:1.49
```
Files with data are created

```
ls -la ec2-tag-compliance 
total 144
drwxrwxr-x 2 wsi wsi   4096 lip  2 13:13 .
drwxrwxr-x 3 wsi wsi   4096 lip  2 13:13 ..
-rw-rw-r-- 1 wsi wsi    127 lip  2 13:13 custodian-run.log
-rw-rw-r-- 1 wsi wsi   1913 lip  2 13:13 metadata.json
-rw-rw-r-- 1 wsi wsi 127990 lip  2 13:13 resources.json
```
Output of custodian report
```
custodian report -s ./report ./custodian_ec2.yml              
CustodianDate,InstanceId,tag:Name,InstanceType,LaunchTime,VpcId,PrivateIpAddress
```

Did I miss something?

**Reply by glauberss2007** [reply]
I am facing the same issue...
